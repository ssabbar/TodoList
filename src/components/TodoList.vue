<template>
  <div class="container">
    <div class="row">
      <div class="col-md-5 mx-auto">
        <h1 class="text-center"> To-Do list</h1>
        <TaskInput/>
        <div class="col-md-10 offset-md-1 mt-2" id="taskslist">
          <h2 class="ml-auto mr-0 my-4">{{completedTaskslength}} out of {{tasks.length}} items completed</h2>
          <TaskList :tasks="tasks"/>
      </div>
      </div>
    </div>
  </div>
</template>

<script>
import TaskList from './TaskList.vue'
import TaskInput from './TaskInput.vue'
export default {
  name: 'TodoList',
  components: {TaskList, TaskInput},
  computed: {
    completedTaskslength () {
      return this.$store.getters.completedTaskslength
    },
    tasks () {
      return this.$store.getters.tasks
    }

  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

</style>
