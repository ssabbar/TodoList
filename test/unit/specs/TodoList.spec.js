import Vue from 'vue'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import TodoList from '@/components/TodoList'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('TodoList component', () => {
    let getters
    let store

    beforeEach(() => {
        getters = {
            tasks: () => { return [{ id: 1, name: 'test', completed: false }] },
            completedTaskslength: () => 1
        }
        store = new Vuex.Store({
            getters
        })
    })

    it('renders correctly', () => {
        const wrapper = mount(TodoList, {
            store,
            localVue
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('display the number of tasks and done tasks correctly', async () => {
        const wrapper = mount(TodoList, {
            store,
            localVue
        })

        const expectedText = getters.completedTaskslength() + ' out of ' + getters.tasks().length + ' items completed'
        expect(wrapper.find('h2').text()).toBe(expectedText)
    })
})

// test('Add task to todo list', async () => {
//   const wrapper = mount(TodoList)

//   expect(wrapper.html()).toContain('0 out of 0 items completed')

//   // test the ADD FUNCTION
//   wrapper.find('input#new-todo-input').setValue('testInput1')
//   await wrapper.find('#addTaskbutton').trigger('click')
//   expect(wrapper.html()).toContain('0 out of 1 items completed')
//   expect(wrapper.html()).toContain('testInput1')

//   wrapper.find('input#new-todo-input').setValue('testInput2')
//   await wrapper.find('#addTaskbutton').trigger('click')
//   expect(wrapper.html()).toContain('0 out of 2 items completed')
//   expect(wrapper.html()).toContain('testInput2')

//   // test the delete FUNCTION
//   await wrapper.find('div#taskid1 button#deleteTaskButton').trigger('click')
//   expect(wrapper.html()).toContain('0 out of 1 items completed')
//   expect(wrapper.html()).not.toContain('testInput1')
//   expect(wrapper.html()).toContain('testInput2')

//   // Test the check function
//   await wrapper.find('div#taskid2 #editInput').trigger('click')
//   expect(wrapper.html()).toContain('1 out of 1 items completed')

//   // test the edit function
//   await wrapper.find('div#taskid2 #editButton').trigger('click')
//   expect(wrapper.html()).toContain('Edit name for testInput2')
//   wrapper.find('input#editInput').setValue('new value')
//   await wrapper.find('#saveEditButton').trigger('click')

//   expect(wrapper.html()).not.toContain('testInput2')
//   expect(wrapper.html()).not.toContain('Edit name for testInput2')
//   expect(wrapper.html()).toContain('new value')

//   // test the cancel of edit task
//   await wrapper.find('div#taskid2 #editButton').trigger('click')
//   expect(wrapper.html()).toContain('Edit name for new value')
//   wrapper.find('input#editInput').setValue('third value')
//   await wrapper.find('#cancelEditButton').trigger('click')

//   expect(wrapper.html()).not.toContain('third value')
//   expect(wrapper.html()).not.toContain('Edit name for third value')
//   expect(wrapper.html()).toContain('new value')
// })
