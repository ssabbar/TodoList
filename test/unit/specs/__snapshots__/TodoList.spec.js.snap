// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component renders correctly 1`] = `
<div class="container">
    <div class="row">
        <div class="col-md-5 mx-auto">
            <h1 class="text-center"> To-Do list</h1>
            <div>
                <div class="text-center mb-2"><label for="new-todo-input">What needs to be done ?</label></div>
                <div class="mb-3"><input id="new-todo-input" class="form-control"></div>
                <div><button id="addTaskbutton" class="btn btn-dark form-control">Add</button></div>
            </div>
            <div id="taskslist" class="col-md-10 offset-md-1 mt-2">
                <h2 class="ml-auto mr-0 my-4">1 out of 1 items completed</h2>
                <div>
                    <div>
                        <div class="mb-4">
                            <div id="taskid1">
                                <div class="mb-2"><input type="checkbox"> <label id="1" class="checkbox-label">test</label></div>
                                <div class="row"><button type="button" id="editTaskButton" class="btn btn-outline-secondary">Edit</button> <button type="button" id="deleteTaskButton" class="btn btn-danger">Delete</button></div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
`;
